code-challenges.test.js

```diff
@@ -44,14 +73,80 @@ const votes2 = { upVotes: 2, downVotes: 33 }
 
 // Pseudo code:
 
+// 1.) Create a function that takes in a parameter(value) 
+
+const voteCounter = (value) => {
+
+// 2.) Have the return value as an agruments and use dot notations to access the values of the upvote and downvote keys and subtract the upvote from the downvote. 
+
+  return value.upVotes - value.downVotes
+}
+
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
+describe("newArray", () => {
+  it("returns one array with no duplicate values.", () => {
+    // const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
+    // const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    expect(dataTypesArray1, dataTypesArray2).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
+  
+    
+
 const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
-// b) Create the function that makes the test pass.
+// // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// // Pseudo code:
+
+// 1.) Create a function with the name of newArray and have the parameter of both arrays.
+
+const newArray = (dataTypesArray1, dataTypesArray2) => {
+
+// 2.) Use the spread method on both data types and return the the array with all the data in the variables.
+
+  return [...dataTypesArray1, ...dataTypesArray2]
+}
+
+// I felt like I had the test right when I started on the method. I tried to use the material LEARN provides and even some google outside sources to get a better understanding on how to combine the arrays and not duplicate. From my understanding, I think that's part of my error at times that it conflicted. I'm still not to saavy understanding how to properly read the errors to understand my next step. I was sucessful using the spread method in my first assessment so I thought that might a better route and I couldnt get the code to work. If I had more time I think I'd try concat() or another method. I typically seen the concat method when searching how to combine 2 array without duplicates.
```

Comment:
The concat method will combine the 2 arrays together, which is exactly what line 114 is doing as well, so you can use either approach to get a combined array. The issue is in your test. Since you're not performing a function call, you're not testing that this function is working as it should be. In the test, you want to be testing that this function is going to output what you're expecting it to output. Without a function call, your test never runs this function to compare the output of this function with the expected output.